### ISSUE TITLE
Need a 32-way decomposition for PEM_Ln9.ne30pg2_EC30to60E2r2.WCYCL1850.pm-cpu_gnu

### ISSUE THREAD

**Issue by ndkeen** [E3SM-Project/E3SM#4964]
I can try adding special pe layouts for pm-cpu, but 32-way seems pretty basic. Can we add this one?

```
ERROR: Could not open block decomposition file for 32 blocks.
CRITICAL ERROR: Filename: /global/cfs/cdirs/e3sm/inputdata/ocn/mpas-o/EC30to60E2r2/mpas-o.graph.info.200904.part.32

```

`PEM_Ln9.ne30pg2_EC30to60E2r2.WCYCL1850.pm-cpu_gnu` is part of `e3sm_integration`

**Comment by ndkeen** [comment]
Ah ok. so maybe it makes more sense to change default PE layout to use, say 128 MPI's so that it would then be looking for a 64-way partition.  In this case, we probably do have enough memory as these nodes have 512GB.  But sounds like it might be better for me to try a change.  Thanks.

**Comment by ndkeen** [comment]
Running with 128 tasks does allow test to pass `PEM_P128_Ln9.ne30pg2_EC30to60E2r2.WCYCL1850.pm-cpu_gnu`.

But after thinking about it more, it would probably be better to have a 32-way partition file as it will fit in memory and running on 1 node would make more sense for testing.

**Comment by jonbob** [comment]
OK, I'll make one -- give me a second

**Comment by ndkeen** [comment]
Thanks.  I first verified I can run a SMS test on 1 node.  And it works, ie:

`SMS_Ln9.ne30pg2_EC30to60E2r2.WCYCL1850.pm-cpu_gnu`

And then I tried using only 32 tasks and that also works:

`SMS_P32_Ln9.ne30pg2_EC30to60E2r2.WCYCL1850.pm-cpu_gnu`

which I assume proves that your new partition files are being used and are OK.

But I get an error (without useful info) when I try `PEM_Ln9.ne30pg2_EC30to60E2r2.WCYCL1850.pm-cpu_gnu`.
It runs the first case, but the `case2` fails.  So it may be some other issue.

**Comment by ndkeen** [comment]
Right, but it's set up to use 64 MPI's per node as default. So the PEM test would use 64 and then 32.

**Comment by jonbob** [comment]
What kind of error do you get? And sorry, I didn't realize how many MPI tasks you were using

**Comment by ndkeen** [comment]
```
 0: MPICH ERROR [Rank 0] [job id 2204497.1] [Thu May 19 10:42:22 2022] [nid005222] - Abort(1734831948) (rank 0 in comm 0): application called MPI_Abort(MPI_COMM_WORLD, 1734831948) - process 0
 0: 
 0: aborting job:
 0: application called MPI_Abort(MPI_COMM_WORLD, 1734831948) - process 0
srun: error: nid005222: task 0: Exited with exit code 255
```

**Comment by jonbob** [comment]
That is useful. Anything in any of the component or cpl logs?

**Comment by ndkeen** [comment]
Ah actually there is something in this file I missed `run/case2run/log.ocean.0000.d0544.err`

```
ERROR: Could not open block decomposition file for 32 blocks.
CRITICAL ERROR: Filename: /global/cfs/cdirs/e3sm/inputdata/ocn/mpas-o/EC30to60E2r2/mpas-o.graph.info.200904.part.32
```

**Comment by ndkeen** [comment]
OK running again, it passes. It might have been that I had 2 cases trying to access that file at the same time while it was being dowloaded.

Closing
